server/src/main/java/org/opensearch/cluster/block/ClusterBlock.java

```diff
@@ -154,24 +175,99 @@ public boolean disableStatePersistence() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        Metadata.XContentContext context = Metadata.XContentContext.valueOf(params.param(CONTEXT_MODE_PARAM, CONTEXT_MODE_API));
         builder.startObject(Integer.toString(id));
         if (uuid != null) {
-            builder.field("uuid", uuid);
+            builder.field(KEY_UUID, uuid);
         }
-        builder.field("description", description);
-        builder.field("retryable", retryable);
+        builder.field(KEY_DESCRIPTION, description);
+        builder.field(KEY_RETRYABLE, retryable);
         if (disableStatePersistence) {
-            builder.field("disable_state_persistence", disableStatePersistence);
+            builder.field(KEY_DISABLE_STATE_PERSISTENCE, disableStatePersistence);
         }
-        builder.startArray("levels");
+        builder.startArray(KEY_LEVELS);
         for (ClusterBlockLevel level : levels) {
             builder.value(level.name().toLowerCase(Locale.ROOT));
         }
         builder.endArray();
+        if (context == Metadata.XContentContext.GATEWAY) {
```

Comment:
Can this cause BWC failures for non-remote cluster?

Reply:
We are only passing GATEWAY as a context when custom metadata should be stored as part of the persistent cluster state. Don't see any reason this could cause any BWC issue.

Reply:
The XContent we are creating is not being sent to the nodes of older version. We will just be uploading this to remote repo. In case of upgrade, we will start publication of cluster state to nodes, when all the nodes have moved to 2.15. So, this would not cause any problem imo

Reply:
Would it not be consumed by the data nodes from remote?